scRNA-seq/01-normalizing_scRNA-seq.Rmd

```diff
@@ -61,28 +62,222 @@ and could use these same steps to filter and normalize the data.
+measured in each cell. 
 
-In this example, we are applying a gene filter cutoff of at least 3 cells 
-expressing each gene.  
-For cell filtering, each cell must express at least 200 genes.
+```{r}
+# Make a vector that contains the number of genes expressed by a particular cell
+num_genes_exp <- apply(detection_mat, 2, sum)
+
+# Let's take a look at what this looks like:
+qplot(num_genes_exp, geom = "density") + 
+  geom_density(fill = "lightblue") +
+  cowplot::theme_cowplot() 
+```
+
+This distribution makes it seem like maybe any cells expressing less than 750
+genes might be a good cutoff?
 
-```{r Set Up Seurat Object}
-# Set up the Seurat object 
-seurat_obj <- CreateSeuratObject(raw.data = sc_data, min.cells = 3,
-                                 min.genes = 200, 
-                                 project = "darmanis_gbm")
+But as you may have noticed in our `sc_metadata` we have cell-type information 
+for these data. 
+Let's use the power of our single-cell resolution data to our advantage and look 
+at this data with cell-type labels. 
+
+```{r}
+# Let's match this to the metadata so we can plot samples by their cell type
+num_genes_exp_df <- data.frame('geo_accession' = colnames(detection_mat), 
+                               num_genes_exp) %>% 
+  dplyr::inner_join(sc_metadata, by = 'geo_accession')
+
+# Plot these data
+ggplot(num_genes_exp_df, aes(x = num_genes_exp)) + 
+  geom_density(fill = "lightblue") +
+  facet_wrap(~cell.type.ch1) +
+  cowplot::theme_cowplot() 
 ```
 
+### Number of samples that express a gene as a quality measure
+
+Now we have an idea of what samples we probably want to get rid of.
+But what if our data contains genes that we can't reliably measure in these cells?
+Some gene types, like many long non-coding RNAs for example, are very tissue 
+specific and wouldn't be useful for us to look at it. 
+For example if only one cell is supposedly expressing a particular gene at a low
+level, do we think that is biologically relevant, or is it more likely a 
+technical artifact? 
+Because our single-cell data is already riddled with zeroes, it's best just to 
+get rid of these genes that we suspect are not being reliably measured by many 
+cells. 
+To do this, we will use our detection matrix to add up how many samples express
+each of these genes. 
+
 ```{r}
-# The filtered count matrix can now be accessed at seurat_obj@raw.data
+# Make a vector that contains the number of cells that express a particular gene
+num_samples_that_exp <- apply(detection_mat, 1, sum)
+
+# Let's take a look at what this looks like:
+qplot(num_samples_that_exp, geom = "density") + 
+  geom_density(fill = "lightblue") +
+  geom_vline(xintercept = 10, col = "red") +
+  cowplot::theme_cowplot() 
+```
+
+### Apply a filter based on our criteria
+
+Now that we've explored our data, let's used the information we've learned to 
+actually filter our data.
+
+```{r Apply filter}
+sc_data_filtered <- GeneMatrixFilter(sc_data_filtered, 
```

Comment:
Wait, so what do you want me to change here?

Reply:
I'm not sure I want you to change anything. Do you think the selection for `num_genes = 200` will have sufficient context?

Reply:
IDK, I honestly don't have super well formed rationale of why 200 as opposed to 100 or 800. So that's why there's not a lot of explanation or context for that number.